CMake/external_pybind11.cmake

```diff
@@ -0,0 +1,65 @@
+cmake_minimum_required(VERSION 3.6)
+include(ExternalProject)
+
+# We use a function to enforce a scoped variables creation only for the build
+# (i.e turn off BUILD_SHARED_LIBS which is used on LRS build as well)
+function(get_pybind11)
+
+    message( STATUS #CHECK_START
+        "Fetching pybind11..." )
+    list( APPEND CMAKE_MESSAGE_INDENT "  " )  # Indent outputs
+
+    # We want to clone the pybind repo and build it here, during configuration, so we can use it.
+    # But ExternalProject_add is limited in that it only does its magic during build.
+    # This is possible in CMake 3.12+ with FetchContent and FetchContent_MakeAvailable in 3.14+ (meaning Ubuntu 20)
+    # but we need to adhere to CMake 3.10 (Ubuntu 18).
+    # So instead, we invoke a new CMake project just to download pybind:
+    configure_file( CMake/pybind11-download.cmake.in
+                    ${CMAKE_BINARY_DIR}/external-projects/pybind11-download/CMakeLists.txt )
+    execute_process( COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+                     WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/external-projects/pybind11-download"
+                     OUTPUT_QUIET )
+    execute_process( COMMAND "${CMAKE_COMMAND}" --build .
+                     WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/external-projects/pybind11-download"
+                     OUTPUT_QUIET )
+
+    # Now that it's available, we can refer to it with an actual ExternalProject_add (but notice we're not
+    # downloading anything)
+    ExternalProject_Add( pybind11
```

Comment:
What will happen here?
I think it will build it again during the build process
Why do we need to build it again?

why not just `pybind11_add_module` as before?

Reply:
as discussed